SafeExamBrowser.SystemComponents/VirtualMachineDetector.cs

```diff
@@ -33,6 +37,17 @@ public bool IsVirtualMachine()
 			isVirtualMachine |= manufacturer.Contains("parallels software");
 			isVirtualMachine |= model.Contains("virtualbox");
 			isVirtualMachine |= manufacturer.Contains("qemu");
+			
+			var macAddr = (from nic in NetworkInterface.GetAllNetworkInterfaces() where nic.OperationalStatus == OperationalStatus.Up select nic.GetPhysicalAddress().ToString()).FirstOrDefault();
+			isVirtualMachine |= ((byte.Parse(macAddr[1].ToString(), NumberStyles.HexNumber) & 2) == 2 || macAddr.StartsWith("080027"));
+
+			using (var searcher = new ManagementObjectSearcher("root\\CIMV2", "SELECT DeviceID FROM Win32_PnPEntity"))
+
+				foreach (ManagementObject queryObj in searcher.Get())
+				{
+				isVirtualMachine |= pciVendorBlacklist.Any(System.Convert.ToString(queryObj.Properties.Cast<PropertyData>().First().Value).ToLower().Contains);
```

Comment:
Please use the property name you are querying for when accessing the object data (see [https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-pnpentity](https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-pnpentity)).